src/webui/api/appcontroller.cpp

```diff
@@ -1207,10 +1213,40 @@ void AppController::getDirectoryContentAction()
         throw APIError(APIErrorType::BadParams, tr("Invalid mode, allowed values: %1").arg(u"all, dirs, files"_s));
     };
 
+    const bool withMetadata {Utils::String::parseBool(params()[u"withMetadata"_s]).value_or(false)};
+
     QJsonArray ret;
     QDirIterator it {dirPath, (QDir::NoDotAndDotDot | parseDirectoryContentMode(visibility))};
     while (it.hasNext())
-        ret.append(it.next());
+    {
+        if (withMetadata)
+        {
+            const QFileInfo fileInfo = it.nextFileInfo();
+            QJsonObject fileObject
+            {
+                {KEY_FILE_METADATA_NAME, fileInfo.fileName()},
+                {KEY_FILE_METADATA_CREATION_DATE, Utils::DateTime::toSecsSinceEpoch(fileInfo.birthTime())},
+                {KEY_FILE_METADATA_LAST_ACCESS_DATE, Utils::DateTime::toSecsSinceEpoch(fileInfo.lastRead())},
+                {KEY_FILE_METADATA_LAST_MODIFICATION_DATE, Utils::DateTime::toSecsSinceEpoch(fileInfo.lastModified())},
+            };
+
+            if (fileInfo.isDir())
+            {
+                fileObject.insert(KEY_FILE_METADATA_TYPE, u"dir"_s);
+            }
+            else if (fileInfo.isFile())
+            {
+                fileObject.insert(KEY_FILE_METADATA_TYPE, u"file"_s);
+                fileObject.insert(KEY_FILE_METADATA_SIZE, fileInfo.size());
```

Comment:
What do you think about the folder size? is it useful to include it?
Just for discussion (no need to implement now)

Reply:
>Folder size is quite complicated because the information isn't available as is, you need to compute it by summing all files and subfolders. It seems quite expensive to implement.

Exactly so I'm more concerned about how users think of it and how useful it is. If it is useful then we might have to implement it eventually (in a another PR).